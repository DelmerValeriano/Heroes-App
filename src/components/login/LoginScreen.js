import { useNavigate } from 'react-router-dom';
import React, { useContext } from 'react'

import { types } from '../../types/types';

import { AuthContext } from '../../auth/authContext';


export const LoginScreen = () => {

  const navigate =useNavigate();
  const {dispatch} =useContext(AuthContext);



  const handleLogin=()=>{

    const actionUser ={
      type: types.login,
      payload :{
        name:'Delmer'
      }
    }


    dispatch(actionUser);

    const lastPath = localStorage.getItem('listPath') || '/marvel';

    navigate(lastPath, {replace:true});
     
  }
  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr/>
      <button 
        className="btn btn-primary"
        onClick={handleLogin}
      >
        Login
      </button>
      
    </div>
  )
}
