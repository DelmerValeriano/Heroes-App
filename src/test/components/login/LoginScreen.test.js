import { mount } from "enzyme";
import { LoginScreen } from "../../../components/login/LoginScreen";
import React from 'react';
import { MemoryRouter, Route,Routes} from 'react-router-dom';
import { AuthContext } from '../../../auth/authContext';
import { types } from '../../../types/types';

const mockNavigate =jest.fn();

jest.mock('react-router-dom',()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate: ()=> mockNavigate,
}))




describe('pruebas en LoginScreen', () => {  
    const ContextValues ={
        dispatch: jest.fn(),
        user:{
            logged: false,
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={ContextValues}>
            <MemoryRouter initialEntries={['/login']}>
                <Routes>
                    <Route path="/" element={<LoginScreen/>}/>
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>     
    );

    test('debe de hacer match con el snapshot', () => {  

        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de realizar el dispatch y la navegacion', () => {  

        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();

        expect(ContextValues.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {name:'Delmer'}
        });

        expect(mockNavigate).toHaveBeenCalledWith('/marvel',{"repalece": true});

        handleClick();
        localStorage.setItem('lasPath','/dc');
        expect(mockNavigate).toHaveBeenCalledWith('/dc',{"repalece": true});

    })

});


