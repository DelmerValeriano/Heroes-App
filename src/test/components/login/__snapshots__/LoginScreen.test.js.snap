// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en LoginScreen debe de hacer match con el snapshot 1`] = `""`;
